### ISSUE TITLE
[Bug]: connect mo failed on mo-cloud that consists of 1cn(tae),1dn(men),3log

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#5696]
### Is there an existing issue for the same bug?

- [ ] #5736

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.6.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

{"level":"ERROR","time":"2022/10/17 16:47:26.474349 +0000","name":"cnservice","caller":"morpc/backend.go:502","msg":"read from backend failed","uuid":"00000000-0000-0000-0000-200000000000","remote":"dn-mem-dn-0.dn-mem-dn-headless.sudong.svc:41010","error":"read tcp4 10.0.118.140:54356->10.0.117.138:41010: use of closed network connection"}
{"level":"ERROR","time":"2022/10/17 16:47:26.474493 +0000","name":"cnservice","caller":"v2@v2.0.3-0.20220926012906-f97d6dabbb6f/session.go:487","msg":"conneciton disconnected","uuid":"00000000-0000-0000-0000-200000000000","remote":"dn-mem-dn-0.dn-mem-dn-headless.sudong.svc:41010","session-id":0}
{"level":"ERROR","time":"2022/10/17 16:47:26.474555 +0000","name":"cnservice","caller":"morpc/backend.go:507","msg":"read loop stopped","uuid":"00000000-0000-0000-0000-200000000000","remote":"dn-mem-dn-0.dn-mem-dn-headless.sudong.svc:41010"}
{"level":"INFO","time":"2022/10/17 16:47:26.474611 +0000","name":"cnservice","caller":"morpc/backend.go:336","msg":"write loop

**Comment by sukki37** [comment]
Additional Information：
1. I can reproduce this locally after restarting mo
```
./mo-service -launch ./etc/launch-tae-CN-mem-DN/launch.toml
ctrl+c
./mo-service -launch ./etc/launch-tae-CN-mem-DN/launch.toml
```
2. connect mo with mysql-client

<img width="918" alt="image" src="https://user-images.githubusercontent.com/77312370/196317919-6716c591-d4ee-43bb-b65c-1ccb3a744e40.png">

**Comment by daviszhen** [comment]
The port is 41010. it is not the 6001.

`{"level":"ERROR","time":"2022/10/17 16:47:26.474349 +0000","name":"cnservice","caller":"morpc/backend.go:502","msg":"read from backend failed","uuid":"00000000-0000-0000-0000-200000000000","remote":"dn-mem-dn-0.dn-mem-dn-headless.sudong.svc:41010","error":"read tcp4 10.0.118.140:54356->10.0.117.138:41010: use of closed network connection"}`

**Comment by aressu1985** [comment]
if port 6001 is not bounded to cn pod id,
mysql client can connect to mo sometimes，but not always

root@test:~/mo-tester# mysql -h dn-mem-tp-cn.sudong  -u dump -p111 -P6001
mysql: [Warning] Using a password on the command line interface can be insecure.
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 1009
Server version: 03 MatrixOne

Copyright (c) 2000, 2022, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> exit
Bye
root@test:~/mo-tester# 
root@test:~/mo-tester# mysql -h dn-mem-tp-cn.sudong  -u dump -p111 -P6001
mysql: [Warning] Using a password on the command line interface can be insecure.

**Comment by zhangxu19830126** [comment]
it seems fixed

**Comment by aressu1985** [comment]
fixed: 249f56400184a0828afdc615c8b5d75e8c57cecc
